### PR TITLE
[Documentation] Update Node Metadata Mgr design w.r.t NCM

### PR DESCRIPTION
**Pull request by pkommoju**

### CODE REVIEW

File: docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+v.02, 2021-03-15
 :toc: right
 
-NOTE: This document is under development
+NOTE: This revision includes only the new API.
 
 == Overview
 
-The node manager is a control plane component responsible of managing and configurating physical servers/nodes/hosts inside a data center.
+The node manager is a control plane component responsible for managing and configuring physical servers/nodes/hosts inside a data center. This is referred to as NMM in this and other documents.
```
Comment (xieus) [inline docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc:11]: Besides physical hosts, we are also starting managing registration of infrastructure components like NCM.

File: docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
+
 It maintains the detailed configuration for each node and collect its health state continuously in a configurable interval.
-The node health states could be used collectively in many user scenarios, for example, determine if a deployment of network resource (for example, port) is successful, or if a load balancing backend is reachable.
+The node health states could be used collectively in many user scenarios, for example, to determine if a deployment of network resource (for example, port) is successful, or if a load balancing backend is reachable.
+
+A user with administrator privileges is assumed to exist (Admin user or just Admin). The Admin interacts with Alcor controller (NodeController/NMM) through an API to register, un-register, query, and update node metadata entries.
```
Comment (xieus) [inline docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc:16]: Very precise 👍

File: docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc [hunk 3]
```diff
@@ -1,19 +1,40 @@
...
+
+NMM API is being extended to interface with the new service called NCM. The new API allow creation, deletion, updating and querying of metadata about NCM entities. An NCM is identified by an ID and a URI. The administrator is responsible to provide a URI for the new NCM being added. There is a provision to specify the maximum number of nodes an NCM is expected to handle, also provided by the Admin.
```
Comment (xieus) [inline docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc:18]: "interface with the new service called NCM" 
Recommend to change to 
=> manage the registration of a new service xref:network_config_manager.adoc[Network Configuration Manager] (NCM)

Comment (xieus) [reply docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc:18]: Note that the NCM design doc is under review and has not been merged to master. Once it is merged, we can click this click and switch to NCM doc.

Comment (pkommoju) [reply docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc:18]: I am updating the document and will push it soon.

File: docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc [hunk 4]
```diff
@@ -35,11 +56,224 @@ The node health states could be used collectively in many user scenarios, for ex
...
+}
+----
+
+==== NMM algorithms
+These outline actions in NMM which are related to NCM only.
+
+===== Node Creation
+** NMM will extract NCM information from the NodeInfo object.
+** Verify that such an NCM exists and adding this node will not exceed the declared capacity of the NCM.
+** Notify the NCM about the node being added to its pool of node.
```
[the remark this answers is not in the document]
Comment (pkommoju) [reply docs/modules/ROOT/pages/mgmt_services/node_metadata_manager.adoc]: Applied all changes and pushed the new revision.

**Comment by xieus** [comment]
MQportcreate.png, vpcGroup.png, and MQ_portcreate.puml files were checked by prior PRs. This is likely caused by the Git issue.

**Comment by pkommoju** [comment]
These changes are moved to a different branch.

**Pull request closed without merging**